filesystem/GD.php

```diff
@@ -390,14 +390,20 @@ public function resizeRatio( $maxWidth, $maxHeight, $useAsMinimum = false ) {
 			return $useAsMinimum ? $this->resizeByWidth( $maxWidth ) : $this->resizeByHeight( $maxHeight );
 	}
 
-	public static function color_web2gd($image, $webColor) {
+	public static function color_web2gd($image, $webColor, $transparencyPercent = 0) {
 		if(substr($webColor,0,1) == "#") $webColor = substr($webColor,1);
 		$r = hexdec(substr($webColor,0,2));
 		$g = hexdec(substr($webColor,2,2));
 		$b = hexdec(substr($webColor,4,2));
 
+		if($transparencyPercent) {
+			if($transparencyPercent > 100) {
+				$transparencyPercent = 100;
+			}
+			$a = 127 * bcdiv($transparencyPercent, 100, 2);
+			return imagecolorallocatealpha($image, $r, $g, $b, $a);
```

Comment:
Actually, you have your scale backwards... imagecolorallocatealpha()'s last parameter is alpha, which denotes opacity, not transparency. A value of 0 transparency should mean opacity (alpha) of 127. Should you rename your method argument to $opacityPercent and set default to 100?

You can also simplify your logic to `if($opacityPercent < 100)` ;p